provider/cmd/app/app.go

```diff
@@ -125,22 +127,27 @@ func (o *Options) MarkFlagsRequired(cmd *cobra.Command) {
 	_ = cmd.MarkFlagRequired("supported-machine-classes")
 }
 
-func Command() *cobra.Command {
+// Command creates cobra command
+// It return pointer to pointer for zap logger because flags and logger are initialized in Execute function
+func Command() (*cobra.Command, **originzap.Logger) {
 	var (
-		zapOpts = zap.Options{Development: true}
```

Comment:
I would control this via a command line flag.

Reply:
Something like this 
```
    var devLogging bool
    flag.BoolVar(&devLogging, "dev-logging", false, "Enable development logging")
    flag.Parse()

    // Setup the logger
    opts := zap.Options{
        Development: devLogging,
    }
    logger := zap.New(zap.UseFlagOptions(&opts))

    // Set the controller-runtime logger to the created logger
    ctrl.SetLogger(logger)
```

Reply:
you can control it with flags (--zap-devel). Flags are bind on bottom of function: https://github.com/ironcore-dev/libvirt-provider/pull/97/files#diff-7c33599ec2dad35992d581529a86b575fe0e5d000ad8d558bdd3afe0688fa423R154

it is only default settings which can be override with flags.

Reply:
technically before it was running default in development mode and you had to put --zap-devel=false for set production mode. I switched logic only, so production mode is default now and you can enable devel mode with --zap-devel.

Reply:
Well the we didn't really had a problem here. I would expect the code being Dev setup first and Prod setup second via config.

Reply:
Ok i can switch it back

Reply:
The default behaviour should be the most simplest setup ever: clone the repo and run the code on your box. Prod setup config is highly custom and we should not make any assumptions here.